docs/transfer-setup.md

```diff
@@ -0,0 +1,281 @@
+
+In this guide, we will cover exporting the key from the Bitcoin node and importing 
+it into the new integrated keyring in the covenant signer. 
+
+## 2. Prerequisites
+
+To successfully complete this guide, ensure the following setup and resources 
+are in place:
+
+1. Setup of bitcoin node in order to retrieve keys. For more information, refer to 
+   [Bitcoin Node Setup](https://github.com/babylonlabs-io/covenant-signer/blob/main/docs/deployment.md#2-bitcoind-setup).
+2. Ensure you have access to the private Bitcoin key, also referred to as the 
+   covenant emulator key, which will be used for signing operations.
+<!-- 3 should we include the babylon node setup? -->
+
+## 3. Install covenant emulator binary 
+
+Download [Golang 1.21](https://go.dev/dl) 
+
+Using the go version 1.21. Once installed run: 
+
+```shell
+go version
+```
+
+Subsequently clone the covenant [repository](https://github.com/babylonlabs-io/covenant-emulator).
+
+```shell
+git clone git@github.com:babylonlabs-io/covenant-emulator.git
+cd covenant-emulator
+git checkout <tag>
+```
+
+Run the following command to build the binaries and
+install them to your `$GOPATH/bin` directory:
+
+```shell
+make install
+```
+
+This command will:
+- Build and compile all Go packages
+- Install `covd` binary to `$GOPATH/bin`
+- Make commands globally accessible from your terminal
+
+If your shell cannot find the installed binary, make sure `$GOPATH/bin` is in
+the `$PATH` of your shell. Use the following command to add it to your profile
+depending on your shell.
+
+```shell
+export PATH=$HOME/go/bin:$PATH
+echo 'export PATH=$HOME/go/bin:$PATH' >> ~/.profile
+```
+
+## 4. Setting up the covenant emulator program
+
+### 4.1. Initialize directories
+
+Next, we initialize the node and home directory. It should generate all of the 
+necessary files such as `covd.config`, these files will live in the `<path>` 
+that you set for the `--home`with the below command.
+
+```shell
+covd init --home <path>
+```
+
+After initialization, the home directory will have the following structure:
+
+```shell
+$ ls <path>
+  ├── covd.conf # Covd-specific configuration file.
+  ├── logs      # Covd logs
+```
+
+### 4.2. Configure the covenant emulator
+
+To set up the connection parameters for the Babylon chain and other covenant 
+emulator settings, configure the `covd.conf` file with the following parameters.
+
+```
+# The interval between each query for pending BTC delegations
+QueryInterval = 15s
+
+# The maximum number of delegations that the covd processes each time
+DelegationLimit = 100
+
+# Bitcoin network to run on
+BitcoinNetwork = simnet
+
+# Babylon specific parameters
+
+# Babylon chain ID
+ChainID = bbn-test-5
+
+# Babylon node RPC endpoint
+RPCAddr = https://rpc-euphrates.devnet.babylonlabs.io:443
+
+# Babylon node gRPC endpoint
+GRPCAddr = https://rpc-euphrates.devnet.babylonlabs.io:443
+
+# Name of the key in the keyring to use for signing transactions
+Key = covenant-key
```

Comment:
so covenant have new config flag which by default is false:
```
; if true, covenant will use the remote signer to sign transactions
RemoteSignerEnabled = false
```
and new config section:
```
[remotesigner]
; URL of the remote signer
URL = http://127.0.0.1:9791

; client when making requests to the remote signer
Timeout = 2s
```
If `RemoteSignerEnabled` is set to true, then the remote signer in `remotesigner` section will be used to sign transaction.

So the whole ordering of the setup looks like follows:
1. First covenant members must setup the covenant signer i.e https://github.com/babylonlabs-io/covenant-emulator/tree/main/covenant-signer. To setup `covenant-signer` covenant members must:
  - export their keys form bitcoind wallet into cosmos keying which will serve as covenant key
  - after importing the keys into the cosmos key ring they must point covenant signer to this key ring i.e setup those field in config: https://github.com/babylonlabs-io/covenant-emulator/blob/main/covenant-signer/example/config.toml#L8
  - after having keyring and config ready they can now launch covenant signer. Note here, covenant signer must be run in some secure network location and be acessible only by covenant emulator
  - when covenant signer is launched, it `url` can be retrieved
  - also before covenant signer can be used for signing, it must be unlocked by using `/v1/unlock` post endpoint with passphrase
2. Now that covenant signer is setup and unlocked its url can be put in covenant emulator config. 
3. The key which covenant emulator specifys in its config is not covenant key, but just key to send cosmos transactions.

Reply:
In general whole architecure will look like this:
![architecture](https://github.com/user-attachments/assets/2b8101f3-b051-4592-8b19-9c2bb5039bd1)
So it is covenant-signer that will have access to covenant-key, and covenant emulator must be the only process allowed to communicate with signer.